### ISSUE TITLE
Support of "WinRT" and "UWP" settings for Visual Studio

### ISSUE THREAD

**Issue by DmitrySokolov** [conan-io/conan#1155]
Visual Studio supports [WinRT](https://en.wikipedia.org/wiki/Windows_Runtime) and [UWP](https://en.wikipedia.org/wiki/Universal_Windows_Platform) application architectures. I think this settings would be useful for many other conan users.

```
    Visual Studio:
        runtime: [MD, MT, MTd, MDd]
        version: ["8", "9", "10", "11", "12", "14", "15"]
        app_arch: ["desktop", "winrt", "uwp"]
```

CMake and some other generators support this settings ([CMake settings for UWP](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-windows-10-universal-applications), [CMake settings for WinRT](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-windows-store)).

**Comment by memsharded** [comment]
But how will these ``app_arch`` values will be mapped to Cmake? from what I see:

```cmake
set(CMAKE_SYSTEM_NAME WindowsStore)
set(CMAKE_SYSTEM_VERSION 10.0)
```
So UWP == WindowsStore? Always?

Also, I cannot see the mapping between ``winrt`` and cmake settings in the cmake help page. Does it means the ``8.1`` version of the WindowsStore?

Now, @lasote is developing some tools for cross-building, that will allow setting those cmake variables in profiles (but not sure about the effect that they should have on settings), see ongoing work here: https://github.com/conan-io/conan/pull/1153

**Comment by DmitrySokolov** [comment]
Yes,
WindowsStore 10.0 == UWP
WindowsStore 8.1 == WinRT

**Comment by memsharded** [comment]
Released in 0.29, feedback welcome
